File: pubnub-gson/pubnub-gson-api/src/main/java/com/pubnub/api/java/models/consumer/datasync/user/DataSyncUser.java
package com.pubnub.api.java.models.consumer.datasync.user;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import lombok.experimental.Accessors;
import org.jetbrains.annotations.Nullable;

import java.util.Map;

/**
 * DataSync User resource returned by {@code pubnub.dataSync().getUser()} / {@code createUser()}.
 *
 * <p>A User is a specialized DataSync entity whose class defaults to {@code User}. The built-in {@code User}
 * class is defined at the {@code GLOBAL} level and its filterable/sortable set is {@code name}
 * ({@code username} / {@code email} are custom-class properties, not built-in {@code User} fields).
 */
@Getter
@Setter
@NoArgsConstructor
@Accessors(chain = true)
@ToString
public class DataSyncUser {
    private String id;
    private String className;
    private int classVersion;

    /**
     * The level at which the entity class is defined (e.g. {@code SubKey} / {@code Global}).
     */
    @Nullable
    private String classLevel;

    private String createdAt;
    private String updatedAt;
    private String eTag;

    @Nullable
    private String status;

    @Nullable
    private String expiresAt;

    @Nullable
    private Map<String, Object> payload;
}

File: pubnub-kotlin/pubnub-kotlin-api/src/commonMain/kotlin/com/pubnub/api/models/consumer/datasync/user/DataSyncUser.kt
package com.pubnub.api.models.consumer.datasync.user

import com.pubnub.api.utils.SerializedName

/**
 * DataSync User resource returned by `pubnub.dataSync.getUser`, `createUser` and referenced by
 * remove operations. A User is a specialized DataSync entity of class `User`.
 *
 * The class-identity properties are exposed with the `class*` names (`className` / `classVersion` /
 * `classLevel`), while the server serializes them under the camelCase `entityClass*` wire keys — hence the
 * `@field:SerializedName` mappings below.
 *
 * @property id User identifier.
 * @property className Entity class identifier (a `User` subclass, defaults to `User`).
 * @property classVersion Version of the entity class.
 * @property classLevel The level at which the entity class is defined (e.g. `SubKey` / `Global`).
 *   Modeled as a nullable [String] rather than an enum to stay source-compatible with future values.
 * @property createdAt Date and time the user was created.
 * @property updatedAt Date and time the user was last updated.
 * @property eTag The user's content fingerprint used in conditional requests.
 * @property status User status.
 * @property expiresAt Date and time when the user expires (will be deleted automatically).
 * @property payload Arbitrary user-defined JSON object.
 */
data class DataSyncUser(
    val id: String,
    @field:SerializedName("entityClass") val className: String,
    @field:SerializedName("entityClassVersion") val classVersion: Int,
    @field:SerializedName("entityClassLevel") val classLevel: String? = null,
    val createdAt: String,
    val updatedAt: String,
    val eTag: String,
    val status: String? = null,
    val expiresAt: String? = null,
    val payload: Map<String, Any?>? = null,
)
